modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/util/ReconnectionUtils.java

```diff
@@ -133,15 +134,21 @@ public static InterceptorChain createReconnectionInterceptorsChain(ExtensionMode
                                                                      ReflectionCache reflectionCache) {
     InterceptorChain.Builder chainBuilder = InterceptorChain.builder();
 
-    if (componentModel instanceof ConnectableComponentModel) {
-      if (((ConnectableComponentModel) componentModel).requiresConnection()) {
-        addConnectionInterceptors(chainBuilder, extensionModel, componentModel, connectionSupplier, reflectionCache);
-      }
+    if (requiresConnectionInterceptors(componentModel)) {
+      addConnectionInterceptors(chainBuilder, extensionModel, componentModel, connectionSupplier, reflectionCache);
```

Comment:
this doesn't look correct. This method adds two interceptors. `ConnectionInterceptor` and `CursorResetInterceptor`. 

The first one you will still need in Mule SDK. At most, you could optimize by removing it when the extension flavour is `in app reuse`

The need for the second should not be determined by the connection provisioning but the presence of `NoReconnectionStrategyModelProperty`. 

The concept of `ConnectionProvisioningModelProperty` makes less and less sense to me

Reply:
Changed to use a flavor-based model property so we can specifically check for the `in app reuse`. Still I kept it encapsulated in a utils method because we need a similar check in more than one place. This way, if we decide to change how we implement the differentiation (flavor or whatever) we don't need to change everywhere (hopefully). Let me know if that makes sense.

Also separated the condition for the `CursorResetInterceptor` so we can check for the `NoReconnectionStrategyModelProperty`.